### PR TITLE
misc updates to build oss docs via docker

### PR DESCRIPTION
**Pull request by lisakowen**
- update Dockerfile and readme
- use ruby 2.5 image
- remove Gemfile.lock

also:
- remove macos-specific doc build instructions

this PR is my attempt at removing the Gemfile.lock from the pxf docs book.  the Dockerfile installs specific versions of some gems, this seemed to fix gem conflict errors that i ran into.  gems are also installed each time you "bundle install".

try it out and let me know if you run into any issues.

### CODE REVIEW

**Review by denalex — APPROVED** [review]

**Review by mperezfuster — APPROVED** [review]
I tried it and it works for me, thanks Lisa!

**Comment by kalensk** [comment]
Can we move this to the main gpdb repo (or some central place)? (I will probably have gpupgarde link to this docs README and Dockerfile for gpupgrade to use. And its making me think it would be best to share across multiple teams and users, etc.)

**Comment by kalensk** [comment]
(Note that I don't really know anything about ruby, gems, mimemagic, bookbinder, etc.)

FWIW when I build the pxf-docs docker image and then run the container to try building the gpupgrade docs I encounter the following issues:


> docker run -it   -p 9292:9292   -p 1234:1234   -v ~/workspace/gp-docs-book-gpupgrade:/gpupgradedocs   --workdir /gpupgradedocs   pxf-docs

**1)** Running it with gpupgrade-docs gives the following error: `bundler: failed to load command: bookbinder (/usr/local/bundle/bin/bookbinder)`

```
 bundle exec bookbinder bind local
/usr/local/bundle/gems/middleman-core-4.1.10/lib/middleman-core/sitemap/resource.rb:88: warning: constant ::Fixnum is deprecated
  skipping (not found) /docs-layout-repo
bundler: failed to load command: bookbinder (/usr/local/bundle/bin/bookbinder)
Bookbinder::Errors::ProgrammerMistake: The method copy_contents cannot copy the contents of the directory '/docs-layout-repo' because it was not found.
  /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/local_filesystem_accessor.rb:62:in `copy_contents'
  /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/components/bind/layout_preparer.rb:19:in `prepare'
  /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/components/bind/directory_preparer.rb:19:in `prepare_directories'
  /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/bind.rb:40:in `run'
  /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/collection.rb:86:in `bind'
  /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/cli.rb:40:in `bind'
  /usr/local/bundle/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
  /usr/local/bundle/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
  /usr/local/bundle/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
  /usr/local/bundle/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
  /usr/local/bundle/gems/bookbindery-10.1.15/install_bin/bookbinder:5:in `<top (required)>'
  /usr/local/bundle/bin/bookbinder:23:in `load'
  /usr/local/bundle/bin/bookbinder:23:in `<top (required)>'
  ```
  
  **2)**  Running bundle install yields the following message:
 
  ```
# bundle install
Fetching gem metadata from http://rubygems.org/......
Your bundle is locked to mimemagic (0.3.2), but that version could not be found in any of the sources listed in your Gemfile. If you haven't changed sources, that means the author
of mimemagic (0.3.2) has removed it. You'll need to update your bundle to a version other than mimemagic (0.3.2) that hasn't been removed in order to install.
  ```
  
 **3)** Even when I try to specify the full path to bookbinder it still complains:
 
 ```
# /usr/local/bundle/gems/bookbindery-10.1.15/install_bin/bookbinder bind local
/usr/local/bundle/gems/middleman-core-4.1.10/lib/middleman-core/sitemap/resource.rb:88: warning: constant ::Fixnum is deprecated
  skipping (not found) /docs-layout-repo
Traceback (most recent call last):
	10: from /usr/local/bundle/gems/bookbindery-10.1.15/install_bin/bookbinder:5:in `<main>'
	 9: from /usr/local/bundle/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	 8: from /usr/local/bundle/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	 7: from /usr/local/bundle/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	 6: from /usr/local/bundle/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	 5: from /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/cli.rb:40:in `bind'
	 4: from /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/collection.rb:86:in `bind'
	 3: from /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/bind.rb:40:in `run'
	 2: from /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/components/bind/directory_preparer.rb:19:in `prepare_directories'
	 1: from /usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/commands/components/bind/layout_preparer.rb:19:in `prepare'
/usr/local/bundle/gems/bookbindery-10.1.15/lib/bookbinder/local_filesystem_accessor.rb:62:in `copy_contents': The method copy_contents cannot copy the contents of the directory '/docs-layout-repo' because it was not found. (Bookbinder::Errors::ProgrammerMistake)
```

**Comment by lisakowen** [comment]
i will chat with @kalensk offline about his issues.

**Comment by kalensk** [comment]
Big thanks to @lisakowen! 🎉  We got it working for gpupgrade and I'll work with @lenapivotal to update the appropriate gpugprade doc README's.

File: concourse/docker/docs/README.md [hunk 1]
```diff
@@ -29,14 +29,21 @@ Once inside the container run the following commands:
...
-rackup
+bundle install
+bundle exec rackup --host=0.0.0.0
 ```
 
 At this point, you can go to a browser window and type
 http://localhost:9292 and you will see the formatted PXF open source
 documentation.
+
+NOTE:  if you perform multiple successive doc builds, remove the generated
```
Comment (dyozie) [inline concourse/docker/docs/README.md:42]: if -> If

**Review by dyozie — APPROVED** [review]

**Comment by lisakowen** [comment]
merging.  will address further comments in a separate PR/commit.

**Pull request merged**